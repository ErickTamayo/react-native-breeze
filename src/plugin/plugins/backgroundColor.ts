import { PluginFunction, PluginPattern } from "./types";

export type PluginGroups = { key: string };

export const pattern: PluginPattern = ({ colorKeys }) => {
  return new RegExp(`^bg-(?<key>${colorKeys("backgroundColor")})$`);
};

export const plugin: PluginFunction<PluginGroups> = ({ groups, color }) => {
  const { key } = groups;
  const value = color("backgroundColor", key);
  return { backgroundColor: value };
};
