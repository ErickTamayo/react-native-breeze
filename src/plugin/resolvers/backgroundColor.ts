import { Resolver, ResolverFunction } from "./types";

export const fn: ResolverFunction = ({ input, groups, theme }) => {
  const { color, number } = groups;

  const colors = theme("colors");
  const colorString = number ? colors[color]?.[number] : colors[color];

  if (typeof colorString !== "string") {
    throw new Error(
      `Invalid color property [${typeof colorString}] for ${input}`
    );
  }

  return { backgroundColor: colorString };
};

export const backgroudColorResolver: Resolver = {
  pattern: /^bg-(?<color>\w+)-?(?<number>\w+)?$/,
  fn,
};
