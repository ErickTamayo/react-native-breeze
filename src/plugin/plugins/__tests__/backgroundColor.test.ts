import {
  shouldEvaluateTheCorrectPatternTest,
  shouldMatchOutputSnapshot,
} from "../../helpers/tests";
import { pattern, plugin } from "../backgroundColor";

describe("backgroundColor", () => {
  const shouldMatch = ["bg-green-500", "bg-black"];

  shouldEvaluateTheCorrectPatternTest(pattern, {
    shouldMatch,
    shouldNotMatch: ["bg-unknown"],
  });

  shouldMatchOutputSnapshot(pattern, plugin, shouldMatch);
});
