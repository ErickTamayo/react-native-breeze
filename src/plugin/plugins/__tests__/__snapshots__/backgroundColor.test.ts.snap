// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`backgroundColor Should parse (bg-black) correctly 1`] = `
Object {
  "backgroundColor": "#000000",
}
`;

exports[`backgroundColor Should parse (bg-green-500) correctly 1`] = `
Object {
  "backgroundColor": "#48bb78",
}
`;
